examples/generic_watcher.rs

```diff
@@ -0,0 +1,39 @@
+// This example shows how to use kube with dynamically known resource kinds.
+
+use color_eyre::Result;
+use futures::prelude::*;
+use kube::{
+    api::{DynamicObject, GroupVersionKind, ListParams, Meta},
+    Api, Client,
+};
+use kube_runtime::{utils::try_flatten_applied, watcher};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug");
+    env_logger::init();
+
+    // alternatively, you can:
+    // 1) take them CLI arguments
+    // 2) use dynamic discovery apis on kube::Client and e.g. watch all
+    // resources in cluster.
+    let group = std::env::var("GROUP").expect("GROUP not set");
+    let group = group.trim();
+    let version = std::env::var("VERSION").expect("VERSION not set");
+    let version = version.trim();
+    let kind = std::env::var("KIND").expect("KIND not set");
+    let kind = kind.trim();
```

Comment:
This example is great. The only thing is probably rename it to dynamic_watcher, the watcher is already generic, technically.
I would probably put the trim on the same chain after env::var.expect if possible, but not important.

Reply:
Unfortunately, it is not possible:
```
temporary value dropped while borrowed
consider using a `let` binding to create a longer lived value
```